libafl/src/events/llmp.rs

```diff
@@ -479,14 +479,15 @@ where
     ///
     /// If the port is not yet bound, it will act as a broker; otherwise, it
     /// will act as a client.
+    /// This will make a new connection to the broker so will return the `client_id` too
     #[cfg(feature = "std")]
     pub fn on_port(
         shmem_provider: SP,
         port: u16,
         configuration: EventConfig,
-    ) -> Result<LlmpEventManager<(), S, SP>, Error> {
-        let llmp = LlmpClient::create_attach_to_tcp(shmem_provider, port)?;
-        Self::new(llmp, configuration)
+    ) -> Result<(LlmpEventManager<(), S, SP>, Option<ClientId>), Error> {
```

Comment:
Maybe the `LlmpEventManager` could simply contain the `ClientId`?
Also, when would it ever be `None`?

Reply:
Actually, `LlmpClient ` could contain the `ClientId`?

Reply:
create_attach_to_tcp can return a client or a broker
if it is a broker then client id is None

Reply:
If it's a client the client should contain the `ClientId` IMHO, so the optional return wouldn't be needed

Reply:
`client.sender.id` already exists!

Reply:
or, in this case, `mgr.llmp.sender.id`

Reply:
it already contains that info.
you can get it by llmp_mgr.sender().id
but there is another id llmp_mgr.receiver().id which is confusing.

so i think it's better that this public API explicitly returns the ID in addition so that user can easily tell which is the right id to use for later use of notify_death

Reply:
Instead, why not add a `client_id()` fn to `LlmpEventManager` that returns the correct one? I don't like an API that returns redundant information and weird optionals

Reply:
Or better yet, just add a function to the event manager that deregisters it from LLMP, and uses the correct id to do so. The user won't need to understand what to do _exactly_ and what IDs to use